zoom-us: use `buildFHSEnv`, update, `xdg-desktop-portal`, minor improvements

Yarny0 opened the pull request:
The most recent [Zoom updates](https://github.com/NixOS/nixpkgs/pull/393649) yield a Zoom package that [fails to function](https://github.com/NixOS/nixpkgs/pull/393649#issuecomment-2782940887).  In a nutshell:

*   The `zoom` program is resistant to `patchelf` even in the old version.  [Some trickery with the dynamic elf loader wrapping `zoom` was required to make it work](https://github.com/NixOS/nixpkgs/pull/381281).
*   The new version's `ZoomLauncher` program is now also "unpatchable" and requires the same trickery.
*   The new version's `ZoomWebviewHost` program is now also "unpatchable", but [I failed to fix it even with wrapping with the elf loader, `LD_LIBRARY_PATH`, and argv0 modifications](https://github.com/NixOS/nixpkgs/pull/393649#issuecomment-2782940887). Note/Reference: [There has been trouble with `ZoomWebviewHost` in the past](https://github.com/NixOS/nixpkgs/issues/344931), but those lessons didn't help me here.

Maybe there is a way to get Zoom working with `patchelf`/`ld.so`/`LD_LIBRARY_PATH` and similar wrappings, but I ultimatelly failed to discover it.  To me, it seems newer Zoom versions insist quite strongly on an FHS-compatible environment.

So the pull request at hand does that: It wraps zoom with `buildFHSEnv`.  With that change, Zoom works without any additional wrappers or (library) path modifications, so most of the patching code used previously disappears.  Besides:

*   Zoom is updated to the current release (by cherry-picking commits from https://github.com/NixOS/nixpkgs/pull/393649 ).
*   Support for `xdg-desktop-portal` is added, fixing https://github.com/NixOS/nixpkgs/issues/359533 .  This [wasn't possible without the FHS environment](https://github.com/NixOS/nixpkgs/pull/381281#issuecomment-2654701506).  However, [that might improve with an upcoming Zoom release](https://github.com/NixOS/nixpkgs/issues/359533#issuecomment-2751724354) (provided that release is usable at all).
*   [The `Exec=` entry in the `.desktop` file is turned into a relative path](https://github.com/NixOS/nixpkgs/issues/308324)
*   Some missing Qt5 libraries are added (discovered with `autoPatchelfHook`).
*   Some `QT5_*` environment variables apparently no longer need to be unset.
*   [Tiny improvements to the test derivation's code](https://github.com/NixOS/nixpkgs/pull/381281#discussion_r1956512725).

Notifying maintainers: @danbst @tadfisher
Notifying @philiptaron per https://github.com/NixOS/nixpkgs/pull/381281#pullrequestreview-2618440109 (thanks again!)

Reference to Zoom meta issue: https://github.com/NixOS/nixpkgs/issues/107495

~~I'm setting this as draft for now as I'm not sure [whether we should wait for the next release](https://github.com/NixOS/nixpkgs/issues/359533#issuecomment-2751724354).~~

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

`nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` wants to rebuild 4k packages and was therefore skipped.

**More**: I prepared the following testing environment:

* x86-64
* built the package as is, i.e., without cherry-picking/rebasing the commit(s) somewhere else
* placed the resulting package into `systemPackages` of
    * an existing and up-to-date NixOS 24.11 installation with X11/Plasma5/Pulseaudio
    * an existing and up-to-date NixOS unstable installation with X11/Xfce/Pulseaudio (inside a VirtualBox)
* Login in X11/Plasma5 with PulseAudio

Tested the following functions successfully:

- [x] sidebar loads (albeit empty on my installation, cf. https://github.com/NixOS/nixpkgs/issues/344931 )
- [x] audio is "detected" by zoom: microphone VU-meter shows amplitude (not tested in VirtualBox)
- [x] audio is produced: "Test speaker" generates audible ting-a-ling (not tested in VirtualBox)
- [x] other participant's audio in video conference is audible (not tested in VirtualBox)
- [x] other participants can hear me (not tested in VirtualBox)
- [x] sending camera picture in video conference works
- [x] participant's camera video is received in video conference
- [x] screen sharing sends screen to other participants
- [x] participant's shared screen is displayed
- [x] "Participants" dialog box is shown (cf. https://github.com/NixOS/nixpkgs/pull/116355 )
- [x] reading team chat messages works as expected
- [x] SSO works: Zoom starts Firefox, then Firefox calls back to Zoom after login, as usual

I did not test the following functions/mechanisms/bindings, but it would probably be good to have them tested:

- [ ] writing team chat messages
- [ ] the sidebar being non-empty
- [ ] `xdg-desktop-portal` (cf. https://github.com/NixOS/nixpkgs/issues/359533 )
- [ ] Pipewire
- [ ] Wayland
- [ ] Darwin

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
zoom-us: 6.3.11.* -> 6.4.1.*

Yarny0 committed:
zoom-us/test: capitalize comments

per https://github.com/NixOS/nixpkgs/pull/381281#discussion_r1956512725

Yarny0 committed:
zoom-us: use `buildFHSEnv` for `x86_64-linux`

While the currently packages Zoom version 6.3.11.7212
is still usabe without FHS environment,
the newer version 6.4.1.587 isn't:
The dynamic loader trick of commit
3f58227a38fa6f3931ef82541572123dc49958f6
can also be applied to `ZoomLauncher`,
making it start again.
However, `ZoomWebviewHost` won't work anymore:
Apparently it also requires the dynamic loader trick;
However, it tries to call itself, calling the dynamic loader,
which then doesn't know what to do.
The result is Zoom causing high CPU load while the main window
is visible, as it endlessly tries to start `ZoomWebviewHost`,
which is apparently needed to show the dashboard.

The commit at hand wraps Zoom in a `buildFHSEnv`.
This is the only possible way to ensure Zoom programs can be
executed without altering them with patchelf or wrapping them.

In detail:

* The derivation that was created before this
  commit is still needed (also for linus system).
  It is now moved into a let-in variable (`unpacked`)
  so it can be used for darwin (still unchanged)
  and for linux (wrapped by FHS env).

  Linux-specific wrapping is removed as we now need the
  zoom directory structure unmodified, as packaged by Zoom.
  In particular, this removes the `postFixup` section,
  or rather, keeps only the darwin-specific part.

  This part of the commit's diff is likely best
  readable with `git diff --color-words=.`.

* As a convenience for later updates, a comment is added that
  might help searching for missing library dependencies.

* The lists of dependencies (`libs` and `binPath`)
  are merged into a function `linuxGetDependencies`.

  It is no longer meaningful to keep them separate as the FHS
  environment merges all those packages into on unified tree.
  It is also not always clear whether a
  package is used as "program" or a "library"
  (e.g. pipewire, or possibly `xdg-desktop-portal`).

  As required by `buildFHSEnv`, the function
  `linuxGetDependencies` is called with `pkgs` as its
  only argument and returns a list of packages that
  will be made available in the FHS environment.
  This mechanism is meant to facilitate multi-arch packages
  (e.g. i686 running on x86-64), so zoom should work alright
  (on x86_64-linux) if we ignore `pkgs` and just provide
  our packages as before; but there is no reason to deviate
  from the mechanism that is provided by `buildFHSEnv`.

r-ryantm committed:
zoom-us: 6.4.1.* -> 6.4.3.*

Yarny0 committed:
zoom-us: add support for `xdg-desktop-portal`

Yarny0 committed:
zoom-us: add Qt5 libraries required by vendored files

Checking library dependencies with `autoPatchelfHook` uncovered
some missing libraries that are required by vendored files.
While zoom seems to work alright without those,
adding them to the LD_LIBRARY_PATH seems to be The Right Thing
and might avoid subtle bugs now or in the future.

Yarny0 committed:
zoom-us: simplify environment

Zoom seems to work fine without

* cd'ing into its `/opt/zoom` directory
* unsetting `QML2_IMPORT_PATH` and `QT_SCREEN_SCALE_FACTORS`

before starting `ZoomLauncher`.

However, unsetting `QT_PLUGIN_PATH` is still required,
at least when Zoom is used within plasma 5.

Yarny0 committed:
zoom-us: use relative path in .desktop file

https://github.com/NixOS/nixpkgs/issues/308324
https://github.com/NixOS/nixpkgs/pull/348601

philiptaron commented:
I think we should drop Darwin support, since the Zoom package for macOS is so much more reliable and I know of no Darwin users. Having this package be more tightly focused on our NixOS usecase, I think, is a really good idea.

@Yarny0, would I prefer that I pushed commits into this PR if I make edits, or open a PR against your nixpkgs?

Yarny0 commented:
> I think we should drop Darwin support, since the Zoom package for macOS is so much more reliable and I know of no Darwin users. Having this package be more tightly focused on our NixOS usecase, I think, is a really good idea.
> 
> @Yarny0, would I prefer that I pushed commits into this PR if I make edits, or open a PR against your nixpkgs?

It's OK if you add commits to the PR at hand (I guess -- I haven't see that in Github, at least not knowingly).  I can also update the branch if you have specific requests.  Would you like me to add a commit to remove Darwin support?  I'd probably add such a commit at the base of the branch to simplify the following commits.

SolidHal commented:
re
> xdg-desktop-portal (cf. https://github.com/NixOS/nixpkgs/issues/359533 )

i tested this package out and was able to screen share on sway out of the box

gvolpe commented:
I tried this out on Hyprland and it works as advertised. The only caveat is that I can only share my entire screen exactly once. If I stop sharing and try to share again, all that's shared is a black screen, so I need to kill zoom and start over.

philiptaron commented:
> I tried this out on Hyprland and it works as advertised. The only caveat is that I can only share my entire screen exactly once. If I stop sharing and try to share again, all that's shared is a black screen, so I need to kill zoom and start over.

Yeah, I have the same issue on GNOME. It's pretty odd and frustrating.

Yarny0 committed:
zoom-us: drop Darwin support

Yarny0 committed:
zoom-us: resolve `with lib;` in `meta`

https://github.com/NixOS/nixpkgs/issues/371862

Yarny0 commented:
*   Rebased onto [current `nixos-unstable`](https://github.com/NixOS/nixpkgs/tree/f675531bc7e6657c10a18b565cfebd8aa9e24c14)
*   In FHS commit: moved `makeWrapper` to Darwin section -- it's no longer needed for linux system.  Also change comment wording/style a bit, and fixed the commit hash in the commit message.
*   Removed Darwin support in a new commit at the head of the branch.  I changed my mind about [placing such a removal at the branch base](https://github.com/NixOS/nixpkgs/pull/397036#issuecomment-2787418178): If it turns out Darwin support is still needed and we have to revert that commit, then it's easier to revert if it's at the head of the branch.

    I'm not sure whether the update script is still correct.  I tested locally by setting the hash and the version to some bogus values, and it worked.

[Tomorrow might bring a new Zoom version](https://github.com/NixOS/nixpkgs/issues/359533#issuecomment-2751724354).  I probably cannot look into it before tuesday evening.

jiriks74 commented:
Seems to be working fine. I'll test it without the portal links I made as a hotfix for the previous version later.

The only thing I don't like is that `hyprland` had to be downloaded and compiled for this but that'll be fine once the resulting portal package is cached upstream.

---

1 package built:
zoom-us

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 397036`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>zoom-us</li>
  </ul>
</details>

Yarny0 commented:
As of this writing, the current Zoom release for Linux is still "6.4.3 (827)", which [I cannot make work properly](https://github.com/NixOS/nixpkgs/pull/393649#issuecomment-2782940887) without `buildFHSEnv`.

So I hereby declare this pull request ready to be reviewed and eventually merged, on which we will have a usable and up-to-date Zoom client in NixOS 25.05.

Note: Besides the tests documented in the PR text, I also quickly tested after merging the branch with the [upcoming `nixos-unstable`](https://github.com/NixOS/nixpkgs/commit/52fa4abe16158f085e3cdf79ce2cbd6fa25bf076) that includes the [latest `staging-next`](https://github.com/NixOS/nixpkgs/pull/395862): It seems to work without further problems.

r-ryantm committed:
zoom-us: 6.4.3.* -> 6.4.5.*

Yarny0 commented:
I tested the [new version](https://github.com/NixOS/nixpkgs/pull/393649#issuecomment-2810197298), but it behaves no different that the previous version:

*   `ZoomLauncher` still won't work after `patchelf` (prints many `XXXXX` and terminates), it needs the elf loader trick
*   `zoom` still won't work after `patchelf` (eats all RAM), it needs the elf loader trick
*   `ZoomWebviewHost` still won't work, no matter what I do, thereby breakting the dashboard and causing high system load as long as the main window is open

Hence, I added the new version to this PR's branch and rebased onto [current `nixos-unstable`](https://github.com/NixOS/nixpkgs/tree/2631b0b7abcea6e640ce31cd78ea58910d31e650).  The result works fine: I repeated all the tests listed in the PR text (with the exception of hearing of and speaking to other participants), and things still work as expected.

r-ryantm committed:
zoom-us: 6.4.5.* -> 6.4.6.*

gvolpe reviewed (approved):
I lost hope in the Zoom team supporting Nix anytime soon, so I think this is the way forward. Thanks for putting up the work to do this!

philiptaron commented on pkgs/by-name/zo/zoom-us/package.nix:
```diff
@@ -2,245 +2,165 @@
+      pkgs.pulseaudio
+    ]
+    ++ lib.optionals xdgDesktopPortalSupport [
+      pkgs.kdePackages.xdg-desktop-portal-kde
+      pkgs.lxqt.xdg-desktop-portal-lxqt
+      pkgs.plasma5Packages.xdg-desktop-portal-kde
+      pkgs.xdg-desktop-portal
+      pkgs.xdg-desktop-portal-gnome
+      pkgs.xdg-desktop-portal-gtk
+      pkgs.xdg-desktop-portal-hyprland
+      pkgs.xdg-desktop-portal-wlr
+      pkgs.xdg-desktop-portal-xapp
```
These are going to make the closure of this package _huge_ -- and it's not clear that any of them are going actually _work_ on NixOS systems.

For instance, I have a GNOME system, so `hyprland`, `kde`, `lxqt`, and so forth are absolutely excess to requirements.

That said, the impact from this bloat is likely minimal.

philiptaron replied:
@spiage here's the list that's causing your ⚠️ bloat indicators ⚠️ to go off.

Yarny0 replied:
It's not that easy as `buildFHSEnv` wants a *function* that picks the right packages out of its sole `pkgs` argument.  Please have a look at https://github.com/NixOS/nixpkgs/pull/403064

philiptaron commented:
I'm going to merge this after giving it a spin on the old computer.

spiage commented:
> I'm going to merge this after giving it a spin on the old computer.

Sir, hello!
Am I really need all of it?
```
Version 33340 -> 33341:
  appstream-qt: +6465.9 KiB
  aquamarine: ∅ → 0.8.0, +1814.6 KiB
  baloo: ∅ → 5.116.0, +3240.7 KiB
  breeze-qt5: ∅ → 5.27.11, +79235.9 KiB
  cinnamon-desktop: ∅ → 6.4.1, +1180.6 KiB
  dconf-editor: ∅ → 45.0.1, +3391.7 KiB
  font-util: ∅ → 1.4.1, +228.7 KiB
  frameworkintegration: +1663.8 KiB
  gcc: +10550.8 KiB
  grim: ∅ → 1.4.1, +49.0 KiB
  gtk-layer-shell: ∅ → 0.9.1, +323.5 KiB
  hyprcursor: ∅ → 0.1.12, +449.5 KiB
  hyprgraphics: ∅ → 0.1.3, +213.1 KiB
  hyprland: ∅ → 0.48.1, +63894.9 KiB
  hyprland-qt-support: ∅ → 0.1.0, +662.4 KiB
  hyprland-qtutils: ∅ → 0.1.3, +661.4 KiB
  hyprlang: ∅ → 0.6.1, +271.3 KiB
  hyprutils: ∅ → 0.6.0, +381.6 KiB
  kcmutils: ∅ → 5.116.0, +1210.5 KiB
  kdeclarative: ∅ → 5.116.0, +1537.8 KiB
  kdecoration: ∅ → 5.27.11, +320.0 KiB
  kdelibs4support: ∅ → 5.116.0, +18752.4 KiB
  kdesignerplugin: ∅ → 5.116.0, +112.4 KiB
  kdesu: ∅ → 5.116.0, +338.6 KiB
  kemoticons: ∅ → 5.116.0, +2682.2 KiB
  kfilemetadata: ∅ → 5.116.0, +1419.4 KiB
  kholidays: ∅ → 5.116.0, +1027.6 KiB
  kidletime: ∅ → 5.116.0, +181.9 KiB
  kinit: ∅ → 5.116.0, +852.1 KiB
  kitemmodels: ∅ → 5.116.0, +561.1 KiB
  kjs: ∅ → 5.116.0, +1213.7 KiB
  kjsembed: ∅ → 5.116.0, +2020.3 KiB
  knewstuff: +1230.9 KiB
  knotifyconfig: ∅ → 5.116.0, +441.4 KiB
  kpackage: +192.8 KiB
  kparts: ∅ → 5.116.0, +977.3 KiB
  kpeople: ∅ → 5.116.0, +630.9 KiB
  kpipewire: ∅ → 5.27.11, +414.5 KiB
  kplotting: ∅ → 5.116.0, +141.6 KiB
  kpty: ∅ → 5.116.0, +197.5 KiB
  kquickcharts: ∅ → 5.116.0, +613.0 KiB
  krunner: ∅ → 5.116.0, +523.2 KiB
  kscreenlocker: ∅ → 5.27.11, +1182.3 KiB
  ktexteditor: ∅ → 5.116.0, +14185.7 KiB
  kunitconversion: ∅ → 5.116.0, +10959.6 KiB
  kuserfeedback: ∅ → 1.3.0, +1809.5 KiB
  kwin: ∅ → 5.27.11, +41992.8 KiB
  kxmlrpcclient: ∅ → 5.116.0, +180.6 KiB
  layer-shell-qt: ∅ → 5.27.11, +135.9 KiB
  libXres: ∅ → 1.2.2, +22.7 KiB
  libdbusmenu-lxqt: ∅ → 0.2.0, +408.3 KiB
  libfm-extra: ∅ → 1.3.2, +49.8 KiB
  libfm-qt: ∅ → 2.1.0, +3534.9 KiB
  libgtop: ∅ → 2.41.3, +602.6 KiB
  libhandy: ∅ → 1.8.3, +1492.3 KiB
  libkscreen: ∅ → 5.27.11, +1479.7 KiB
  libksysguard: ∅ → 5.27.11, +8097.1 KiB
  libmatekbd: ∅ → 1.28.0, +841.9 KiB
  libmatemixer: ∅ → 1.28.0, +934.6 KiB
  libmateweather: ∅ → 1.28.0, +115748.4 KiB
  libqaccessibilityclient: +286.7 KiB
  libqtxdg: ∅ → 4.1.0, +623.2 KiB
  libspng: ∅ → 0.7.4, +101.7 KiB
  libwnck: ∅ → 43.2, +2137.7 KiB
  libxcb: +1149.4 KiB
  lxqt-qtplugin: ∅ → 2.1.0, +204.0 KiB
  marco: ∅ → 1.28.1, +6840.5 KiB
  mate-desktop: ∅ → 1.28.2, +3411.8 KiB
  mate-menus: ∅ → 1.28.0, +659.9 KiB
  mate-panel: ∅ → 1.28.4, +14998.0 KiB
  mate-settings-daemon: ∅ → 1.28.0, +3693.4 KiB
  menu-cache: ∅ → 1.1.0, +165.0 KiB
  networkmanager-qt: ∅ → 5.116.0, +2312.0 KiB
  nixos-system-a7: 25.05.20250429.aa3b7a0 → 25.05.20250429.f29c42a
  nss: +62.0 KiB
  pkgconf: ∅ → 2.4.3, +183.5 KiB
  pkgconf-wrapper: ∅ → 2.4.3, +11.4 KiB
  plasma-framework: ∅ → 5.116.0, +8848.9 KiB
  plasma-workspace: ∅ → 5.27.11.1, +63036.9 KiB
  poppler-qt5: ∅ → 24.02.0, +5813.9 KiB
  prison: ∅ → 5.116.0, +268.8 KiB
  python3: +282.3 KiB
  qqc2-desktop-style: ∅ → 5.116.0, +458.3 KiB
  qt3d: ∅ → 5.15.16, +17915.1 KiB
  qtbase: +27419.2 KiB
  qtdeclarative: +7549.0 KiB
  qtgamepad: ∅ → 5.15.16, +344.3 KiB
  qtgraphicaleffects: ∅ → 5.15.16, +583.0 KiB
  qtlocation: ∅ → 5.15.16, +14257.9 KiB
  qtlottie: ∅ → 5.15.16, +416.5 KiB
  qtremoteobjects: ∅ → 5.15.16, +1709.9 KiB
  qtsensors: ∅ → 5.15.16, +1099.9 KiB
  qtsvg: +91.9 KiB
  qtvirtualkeyboard: ∅ → 5.15.16, +5918.1 KiB
  qtwebchannel: ∅ → 5.15.16, +245.4 KiB
  qtwebengine: ∅ → 5.15.18, +230856.6 KiB
  qtxmlpatterns: ∅ → 5.15.16, +5592.4 KiB
  re2: ∅ → 2024-07-02, +596.8 KiB
  sdbus-cpp: ∅ → 2.0.0, +940.6 KiB
  seatd: ∅ → 0.9.1, +81.4 KiB
  slurp: ∅ → 1.5.0, +55.5 KiB
  threadweaver: ∅ → 5.116.0, +265.1 KiB
  tomlplusplus: ∅ → 3.4.0, +1218.2 KiB
  wayland: +42.5 KiB
  webp-pixbuf-loader: ∅ → 0.2.6, +35.2 KiB
  xapp: ∅ → 2.8.8, +1048.0 KiB
  xcb-util-errors: ∅ → 1.0.1, +47.4 KiB
  xdg-desktop-portal-gnome: ∅ → 47.3, +1436.8 KiB
  xdg-desktop-portal-hyprland: ∅ → 1.3.9, +1051.6 KiB
  xdg-desktop-portal-kde: ∅ → 5.27.11, +1941.4 KiB
  xdg-desktop-portal-lxqt: ∅ → 1.1.0, +269.0 KiB
  xdg-desktop-portal-wlr: ∅ → 0.7.1, +140.5 KiB
  xdg-desktop-portal-xapp: ∅ → 1.1.0, +402.3 KiB
  xwayland: +2683.8 KiB
  zoom: 6.3.11.7212 → 6.4.6.1370, 6.4.6.1370-fhsenv, +32467.9 KiB
<<< /run/booted-system
>>> /run/current-system
Version changes:
[C.]  #01  appstream-qt              1.0.4 -> 1.0.4 x2
[C*]  #02  baloo                     6.13.0 -> 5.116.0, 6.13.0
[U*]  #03  fastfetch                 2.41.0, 2.41.0-man -> 2.42.0, 2.42.0-man
[C.]  #04  fix-qt-builtin-paths.sh   <none> -> <none> x2
[C.]  #05  fix-qt-module-paths.sh    <none> -> <none> x2
[C*]  #06  frameworkintegration      5.116.0, 6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #07  gcc                       14.2.1.20250322, 14.2.1.20250322-lib, 14.2.1.20250322-libgcc -> 14.2.1.20250322, 14.2.1.20250322-lib x2, 14.2.1.20250322-libgcc x2, 14.2.1.20250322-man
[C.]  #08  gdk-pixbuf-loaders.cache  <none> -> <none> x2
[C*]  #09  kcmutils                  6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #10  kdeclarative              6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #11  kdecoration               6.3.4 -> 5.27.11, 6.3.4
[C.]  #12  kdesu                     6.13.0 -> 5.116.0, 6.13.0
[C*]  #13  kfilemetadata             6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #14  kholidays                 6.13.0 -> 5.116.0, 6.13.0
[C.]  #15  kidletime                 6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #16  kitemmodels               6.13.0 -> 5.116.0, 6.13.0
[C.]  #17  knewstuff                 5.116.0, 6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #18  knotifyconfig             6.13.0 -> 5.116.0, 6.13.0
[C*]  #19  kpackage                  5.116.0, 6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #20  kparts                    6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #21  kpeople                   6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #22  kpipewire                 6.3.4 -> 5.27.11, 6.3.4
[C.]  #23  kplotting                 6.13.0 -> 5.116.0, 6.13.0
[C.]  #24  kpty                      6.13.0 -> 5.116.0, 6.13.0
[C.]  #25  kquickcharts              6.13.0 -> 5.116.0, 6.13.0
[C*]  #26  krunner                   6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C*]  #27  kscreenlocker             6.3.4 -> 5.27.11, 6.3.4
[C*]  #28  ktexteditor               6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #29  kunitconversion           6.13.0 -> 5.116.0, 6.13.0
[C.]  #30  kuserfeedback             6.13.0 -> 1.3.0, 6.13.0
[C*]  #31  kwin                      6.3.4 -> 5.27.11, 6.3.4
[C.]  #32  layer-shell-qt            6.3.4 -> 5.27.11, 6.3.4
[C*]  #33  libkscreen                6.3.4 -> 5.27.11, 6.3.4
[C*]  #34  libksysguard              6.3.4 -> 5.27.11, 5.27.11-bin, 6.3.4
[C.]  #35  libqaccessibilityclient   0.6.0 -> 0.6.0 x2
[C.]  #36  libxcb                    1.17.0, 1.17.0-dev -> 1.17.0, 1.17.0-dev, 1.17.0-man
[C.]  #37  networkmanager-qt         6.13.0 -> 5.116.0, 6.13.0
[D.]  #38  nixos-system-a7           25.05.20250429.8068b83 -> 25.05.20250429.f29c42a
[C.]  #39  nss                       3.101.2 -> 3.101.2, 3.101.2-man
[C*]  #40  plasma-workspace          6.3.4 -> 5.27.11.1, 6.3.4
[C.]  #41  prison                    6.13.0 -> 5.116.0, 6.13.0
[C*]  #42  python3                   3.11.11, 3.12.9 x2, 3.12.9-env x4 -> 3.11.11, 3.12.9 x2, 3.12.9-env x5
[C*]  #43  qqc2-desktop-style        6.13.0 -> 5.116.0, 5.116.0-bin, 6.13.0
[C.]  #44  qtbase                    5.15.16, 5.15.16-bin, 6.9.0, 6.9.0-dev, 6.9.0-only-plugins-qml -> 5.15.16, 5.15.16-bin, 5.15.16-dev, 6.9.0, 6.9.0-dev, 6.9.0-only-plugins-qml
[C.]  #45  qtdeclarative             5.15.16, 5.15.16-bin, 6.9.0, 6.9.0-dev -> 5.15.16, 5.15.16-bin, 5.15.16-dev, 6.9.0, 6.9.0-dev
[C.]  #46  qtlocation                6.9.0 -> 5.15.16, 5.15.16-bin, 6.9.0
[C.]  #47  qtsensors                 6.9.0 -> 5.15.16, 5.15.16-bin, 6.9.0
[C*]  #48  qtsvg                     5.15.16, 5.15.16-bin, 6.9.0, 6.9.0-dev -> 5.15.16, 5.15.16-bin, 5.15.16-dev, 6.9.0, 6.9.0-dev
[C.]  #49  qtvirtualkeyboard         6.9.0 -> 5.15.16, 6.9.0
[C.]  #50  qtwebchannel              6.9.0 -> 5.15.16, 5.15.16-bin, 6.9.0
[C.]  #51  qtwebengine               6.9.0 -> 5.15.18, 5.15.18-bin, 6.9.0
[C.]  #52  threadweaver              6.13.0 -> 5.116.0, 6.13.0
[C.]  #53  wayland                   1.23.1, 1.23.1-dev -> 1.23.1, 1.23.1-dev, 1.23.1-man
[C*]  #54  xdg-desktop-portal-kde    6.3.4 -> 5.27.11, 6.3.4
[C*]  #55  xwayland                  24.1.6 -> 24.1.6 x2
[U*]  #56  zoom                      6.3.11.7212 -> 6.4.6.1370 x2, 6.4.6.1370-bwrap, 6.4.6.1370-fhsenv-profile, 6.4.6.1370-fhsenv-rootfs, 6.4.6.1370-init
Added packages:
[A.]  #01  aquamarine                   0.8.0
[A.]  #02  breeze-qt5                   5.27.11, 5.27.11-bin
[A.]  #03  cinnamon-desktop             6.4.1
[A.]  #04  dconf-editor                 45.0.1
[A+]  #05  etcd                         3.5.19
[A.]  #06  etcdctl                      3.5.19
[A.]  #07  etcdserver                   3.5.19
[A.]  #08  etcdutl                      3.5.19
[A.]  #09  font-util                    1.4.1
[A.]  #10  grim                         1.4.1
[A.]  #11  gtk-layer-shell              0.9.1
[A.]  #12  hyprcursor                   0.1.12-lib
[A.]  #13  hyprgraphics                 0.1.3
[A.]  #14  hyprland                     0.48.1
[A.]  #15  hyprland-qt-support          0.1.0
[A.]  #16  hyprland-qtutils             0.1.3
[A.]  #17  hyprlang                     0.6.1
[A.]  #18  hyprutils                    0.6.0
[A.]  #19  kdelibs4support              5.116.0
[A.]  #20  kdesignerplugin              5.116.0
[A.]  #21  kemoticons                   5.116.0, 5.116.0-bin
[A.]  #22  kinit                        5.116.0
[A.]  #23  kjs                          5.116.0
[A.]  #24  kjsembed                     5.116.0
[A.]  #25  kxmlrpcclient                5.116.0
[A.]  #26  libXres                      1.2.2
[A.]  #27  libdbusmenu-lxqt             0.2.0
[A.]  #28  libfm-extra                  1.3.2
[A.]  #29  libfm-qt                     2.1.0
[A.]  #30  libgtop                      2.41.3
[A.]  #31  libhandy                     1.8.3
[A.]  #32  libmatekbd                   1.28.0
[A.]  #33  libmatemixer                 1.28.0
[A.]  #34  libmateweather               1.28.0
[A.]  #35  libqtxdg                     4.1.0
[A.]  #36  libspng                      0.7.4
[A.]  #37  libwnck                      43.2
[A.]  #38  lxqt-qtplugin                2.1.0
[A.]  #39  marco                        1.28.1
[A.]  #40  mate-desktop                 1.28.2
[A.]  #41  mate-menus                   1.28.0
[A.]  #42  mate-panel                   1.28.4
[A.]  #43  mate-settings-daemon         1.28.0
[A.]  #44  menu-cache                   1.1.0
[A.]  #45  pkgconf                      2.4.3, 2.4.3-lib
[A.]  #46  pkgconf-wrapper              2.4.3
[A.]  #47  plasma-framework             5.116.0, 5.116.0-bin
[A.]  #48  poppler-qt5                  24.02.0
[A.]  #49  qt3d                         5.15.16, 5.15.16-bin
[A.]  #50  qtgamepad                    5.15.16, 5.15.16-bin
[A.]  #51  qtgraphicaleffects           5.15.16
[A.]  #52  qtlottie                     5.15.16
[A.]  #53  qtremoteobjects              5.15.16
[A.]  #54  qtxmlpatterns                5.15.16
[A.]  #55  re2                          2024-07-02
[A.]  #56  sdbus-cpp                    2.0.0
[A.]  #57  seatd                        0.9.1
[A.]  #58  slurp                        1.5.0
[A.]  #59  tomlplusplus                 3.4.0
[A.]  #60  webp-pixbuf-loader           0.2.6
[A.]  #61  xapp                         2.8.8
[A.]  #62  xcb-util-errors              1.0.1
[A.]  #63  xdg-desktop-portal-gnome     47.3
[A.]  #64  xdg-desktop-portal-hyprland  1.3.9
[A.]  #65  xdg-desktop-portal-lxqt      1.1.0
[A.]  #66  xdg-desktop-portal-wlr       0.7.1
[A.]  #67  xdg-desktop-portal-xapp      1.1.0
Closure size: 2473 -> 2620 (248 paths added, 101 paths removed, delta +147, disk usage +931.6MiB).
```

philiptaron commented:
I'll gladly merge changes which reduce the default closure size without impeding the Zoom functionality. If you don't use `xdg-desktop` integration, one fast closure reduction is to override with `xdgDesktopPortalSupport = false`.

spiage commented:
> I'll gladly merge changes which reduce the default closure size without impeding the Zoom functionality. If you don't use `xdg-desktop` integration, one fast closure reduction is to override with `xdgDesktopPortalSupport = false`.

Thank you!
This way is better )
```
    (pkgs.zoom-us.override { xdgDesktopPortalSupport = false; }) # zoom-us # zoom.us video conferencing application
```

```
<<< /run/booted-system
>>> /run/current-system
Version changes:
[U*]  #01  fastfetch                2.41.0, 2.41.0-man -> 2.42.0, 2.42.0-man
[C.]  #02  fix-qt-builtin-paths.sh  <none> -> <none> x2
[C.]  #03  fix-qt-module-paths.sh   <none> -> <none> x2
[C.]  #04  gcc                      14.2.1.20250322, 14.2.1.20250322-lib, 14.2.1.20250322-libgcc -> 14.2.1.20250322, 14.2.1.20250322-lib, 14.2.1.20250322-libgcc, 14.2.1.20250322-man
[C.]  #05  libxcb                   1.17.0, 1.17.0-dev -> 1.17.0, 1.17.0-dev, 1.17.0-man
[U.]  #06  nixos-system-a7          25.05.20250429.8068b83 -> 25.05.20250430.ac95911
[C.]  #07  nss                      3.101.2 -> 3.101.2, 3.101.2-man
[C.]  #08  qtbase                   5.15.16, 5.15.16-bin, 6.9.0, 6.9.0-dev, 6.9.0-only-plugins-qml -> 5.15.16, 5.15.16-bin, 5.15.16-dev, 6.9.0, 6.9.0-dev, 6.9.0-only-plugins-qml
[C.]  #09  qtdeclarative            5.15.16, 5.15.16-bin, 6.9.0, 6.9.0-dev -> 5.15.16, 5.15.16-bin, 5.15.16-dev, 6.9.0, 6.9.0-dev
[C*]  #10  qtsvg                    5.15.16, 5.15.16-bin, 6.9.0, 6.9.0-dev -> 5.15.16, 5.15.16-bin, 5.15.16-dev, 6.9.0, 6.9.0-dev
[C.]  #11  wayland                  1.23.1, 1.23.1-dev -> 1.23.1, 1.23.1-dev, 1.23.1-man
[U*]  #12  zoom                     6.3.11.7212 -> 6.4.6.1370 x2, 6.4.6.1370-bwrap, 6.4.6.1370-fhsenv-profile, 6.4.6.1370-fhsenv-rootfs, 6.4.6.1370-init
Added packages:
[A+]  #1  etcd             3.5.19
[A.]  #2  etcdctl          3.5.19
[A.]  #3  etcdserver       3.5.19
[A.]  #4  etcdutl          3.5.19
[A.]  #5  qt3d             5.15.16, 5.15.16-bin
[A.]  #6  qtgamepad        5.15.16, 5.15.16-bin
[A.]  #7  qtlottie         5.15.16
[A.]  #8  qtremoteobjects  5.15.16
[A.]  #9  qtxmlpatterns    5.15.16
Closure size: 2473 -> 2498 (126 paths added, 101 paths removed, delta +25, disk usage +155.1MiB).
```

booxter commented:
> I think we should drop Darwin support, since the Zoom package for macOS is so much more reliable and I know of no Darwin users

Could anyone please explain what this Zoom package that is more reliable is? I used zoom-us with no problems until now, so I'd like to understand how I can revive the build for Darwin; or migrate to something else. Thanks.

booxter commented:
@philiptaron There are a number of packages in tree that do if-else for platforms (like Obsidian). No reason to have it as a separate package.

philiptaron commented:
Yes, I don't mean under a different name, rather as darwin.nix in this directory under a top-level if statement. What I mean by "separate package" is that the method of packaging that works on Linux — buildFHSEnv — likely isn't right for Darwin. It'll need some other way of packaging. Feel free to find that way out.

booxter commented:
FWIW the Darwin implementation that was present before the drop worked fine. On MacOS, it's not complicated: you just unpack a package and copy `Applications/*` to `$out`.

I can craft something to restore the support, but I'd like to understand what the original motivation for the drop was. Is it that you personally don't want to deal with the complexity of supporting two platforms? Or was something actually broken?

winterqt commented:
Yeah, I don’t see any harm in keeping the Darwin codepath under the same attrname — it’s a pretty common pattern.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/issues-launching-zoom-on-gnome/69923/1
